src/screens/Messages/Conversation/MessagesList.tsx

```diff
@@ -264,20 +269,71 @@ export function MessagesList({
   // -- Message sending
   const onSendMessage = useCallback(
     async (text: string) => {
-      let rt = new RichText({text}, {cleanNewlines: true})
-      await rt.detectFacets(agent)
-      rt = shortenLinks(rt)
+      let rt = new RichText({text: text.trimEnd()}, {cleanNewlines: true})
+
+      // detect facets without resolution first - this is used to see if there's
+      // any post links in the text that we can embed. We do this first because
+      // we want to remove the post link from the text, re-trim, then detect facets
+      rt.detectFacetsWithoutResolution()
+
+      let embed: AppBskyEmbedRecord.Main | undefined
+      // find the first link facet that is a link to a post
+      const postLinkFacet = rt.facets?.find(facet => {
+        return facet.features.find(feature => {
+          if (AppBskyRichtextFacet.isLink(feature)) {
+            return isBskyPostUrl(feature.uri)
+          }
+          return false
+        })
+      })
 
-      // filter out any mention facets that didn't map to a user
-      rt.facets = rt.facets?.filter(facet => {
-        const mention = facet.features.find(feature =>
-          AppBskyRichtextFacet.isMention(feature),
+      // if we found a post link, get the post and embed it
+      if (postLinkFacet) {
+        const postLink = postLinkFacet.features.find(
+          AppBskyRichtextFacet.isLink,
         )
-        if (mention && !mention.did) {
-          return false
+        if (!postLink) return
+
+        try {
+          const post = await getPostAsQuote(getPost, postLink.uri)
+          if (post) {
+            embed = {
+              $type: 'app.bsky.embed.record',
+              record: {
+                uri: post.uri,
+                cid: post.cid,
+              },
+            }
+
+            // remove the post link from the text
+            rt.delete(
+              postLinkFacet.index.byteStart,
+              postLinkFacet.index.byteEnd,
+            )
+
+            // re-trim the text, now that we've removed the post link
+            //
+            // if the post link is at the start of the text, we don't want to leave a leading space
+            // so trim on both sides
+            if (postLinkFacet.index.byteStart === 0) {
+              rt = new RichText({text: rt.text.trim()}, {cleanNewlines: true})
+            } else {
+              // otherwise just trim the end
+              rt = new RichText(
+                {text: rt.text.trimEnd()},
+                {cleanNewlines: true},
+              )
+            }
```

Comment:
Hadn't thought hard about this before, but we were using `trimEnd` when sending, and `trim` inside `Convo`.

Do we want to allow leading whitespace, or should we just do `trim` in all cases?

Reply:
I was leaning towards `trimEnd` just to match how the main composer does it, but I'm not particularly fussed which way we do it